suitcase/fields.py

```diff
@@ -656,6 +725,51 @@ def setval(self, value):
         return self._get_parent_field().setval(value)
 
 
+class FieldArray(BaseField):
+    """Field which contains a list of some other field.
+
+    In some protocols, there exist repeated substructures which are present in a
+    variable number. The variable nature of these fields make a DispatchField
+    and DispatchTarget combination unsuitable; instead a FieldArray may be
+    used to pack/unpack these fields to/from a list.
+

```

Comment:
Come to think of it, I have never embedded a structure in this way on a project, so it is possible that this just doesn't work right now (and probably should).  I think that non-fields that get added will just be ignored, so we would need to do something like the following to get just one.

```python
class FamousQuoteRecord(Structure):
    length = LengthField(UBInt16())
    value = Payload(length)
    author = SubstructureField(PascalString16())
```
Or, maybe we could do something like...

```python
author = PascalString16.to_field()
```

Or 

```python
PascalStringField = make_structure_field(PascalString16)

# ...
```